python/pyspark/ml/tests.py

```diff
@@ -112,14 +112,15 @@ def test_pipeline(self):
         self.assertEqual(6, dataset.index)
 
 
-class TestParams(HasMaxIter, HasInputCol):
+class TestParams(HasMaxIter, HasInputCol, HasSeed):
     """
-    A subclass of Params mixed with HasMaxIter and HasInputCol.
+    A subclass of Params mixed with HasMaxIter, HasInputCol and HasSeed.
     """
 
-    def __init__(self):
+    def __init__(self, seed=None):
         super(TestParams, self).__init__()
         self._setDefault(maxIter=10)
+        self._set(seed=seed)
```

Comment:
Yes, that is the desired effect. If in the kwargs it is set to one then we use the default. This also matches the behaviour taken in the classes that use hasSeed (although they do wrap it through a few extra levels of function calls I left off here for test simplicity).